### PR TITLE
Interaction rightOver

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/Ui/WindowManager.h [hunk 1]
```diff
@@ -135,7 +135,9 @@ namespace OpenLoco::Ui::Windows
         void setToRoadExtra(Window* main, Map::RoadElement* track, const uint8_t bh, const Map::Pos2 pos);
         void sub_4A6FAC();
         bool isStationTabOpen();
-        bool rotate(Window* self);
+        bool isOverheadTabOpen();
+        bool isSignalTabOpen();
+        void rotate(Window* self);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Ui/WindowManager.h:140]: bool

File: src/OpenLoco/Map/Tile.h [hunk 2]
```diff
@@ -182,6 +176,12 @@ namespace OpenLoco::Map
             _type |= state ? 0x40 : 0;
         }
         void createWave(int16_t x, int16_t y, int animationIndex);
+        bool hasHighTypeFlag() const { return _type & 0x80; }
+        void setHighTypeFlag(bool state)
+        {
+            _type &= ~0x80;
+            _type |= state ? 0x80 : 0;
+        }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Map/Tile.h:184]: Why did you remove these members from `TileElementBase` to `SurfaceElement`?

File: src/OpenLoco/Ui/ViewportInteraction.cpp [hunk 3]
```diff
@@ -401,6 +407,467 @@ namespace OpenLoco::Ui::ViewportInteraction
...
+            return false;
+        }
+        auto* wallObj = ObjectManager::get<WallObject>(wall->wallObjectId());
+        FormatArguments::mapToolTip(StringIds::stringid_right_click_to_remove, wallObj->name);
+        return true;
+    }
+
+    // 0x004CDE78
+    // Note: this is only when in a constructing mode
+    static bool rightOverBuildingConstruct(InteractionArg& interaction)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Ui/ViewportInteraction.cpp:731]: On this PR, buildings are right-click removable outside of construction mode. I can't reproduce this on master (c8228154).

File: src/OpenLoco/Ui/ViewportInteraction.cpp [hunk 4]
```diff
@@ -428,20 +895,82 @@ namespace OpenLoco::Ui::ViewportInteraction
...
+                break;
+        }
+        if (hasInteraction)
+        {
+            return interaction;
+        }
+        else
+        {
+            return InteractionArg{};
+        }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Ui/ViewportInteraction.cpp:973]: This is down to personal preference, but if you like:
```suggestion
        return hasInteraction ? interaction : InteractionArg{};
```

File: src/OpenLoco/Map/Tile.h (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Map/Tile.h:184]: They weren't generic to all tile elements so i moved them to the specific instance.

**Review by AaronVanGeffen — APPROVED** [review]
